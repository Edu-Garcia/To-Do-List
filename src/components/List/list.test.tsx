import { render } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom';
import { List } from '.'
import '@testing-library/jest-dom';

describe('List Tasks Tests', () => {
  it('if you have an item in the list, show the list', () => {
    const mockTasks = [
      {
        id: '1',
        title: 'Tarefa 1',
        description: 'Descrição da tarefa 1',
        complete: false,
        user: null
      }
    ]

    const { container } = render(
      <BrowserRouter>
        <List
          tasks={mockTasks}
          completeTask={jest.fn()}
          deleteTask={jest.fn()}
          editTask={jest.fn()}
          pendingFilter={false}
          completeFilter={false}
        />
      </BrowserRouter>
    );

    expect(container).toMatchSnapshot();
  });

  it('if you do not have any item in the list, show a text informing', () => {
    const { container } = render(
      <BrowserRouter>
        <List
          tasks={[]}
          completeTask={jest.fn()}
          deleteTask={jest.fn()}
          editTask={jest.fn()}
          pendingFilter={false}
          completeFilter={false}
        />
      </BrowserRouter>
    );

    expect(container).toMatchSnapshot();
  });
});