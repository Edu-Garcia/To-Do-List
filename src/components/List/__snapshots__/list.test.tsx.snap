// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Tasks Tests if you do not have any item in the list, show a text informing 1`] = `
<div>
  <div
    class="emptyList"
  >
    <h2>
      Nenhuma tarefa encontrada
    </h2>
  </div>
</div>
`;

exports[`List Tasks Tests if you have an item in the list, show the list 1`] = `
<div>
  <ul>
    <li
      class=""
    >
      <div
        class="textContainer"
      >
        <h3>
          Tarefa 1
        </h3>
        <span>
          Descrição da tarefa 1
        </span>
      </div>
      <button
        class="button undefined"
        type="button"
      >
        Concluir
      </button>
      <button
        class="button edit"
        type="button"
      >
        Editar
      </button>
      <button
        class="button delete"
        type="button"
      >
        Excluir
      </button>
    </li>
  </ul>
</div>
`;
